lib/src/test/java/com/thenewboston/utils/BankApiMockEngine.kt

```diff
@@ -23,90 +36,101 @@ class BankApiMockEngine {
-                        val content = BankAPIJsonMapper.mapValidatorToJson()
-                        if (enableErrorResponse) {
-                            respond(errorContent, HttpStatusCode.InternalServerError, responseHeaders)
-                        } else {
-                            respond(content, HttpStatusCode.OK, responseHeaders)
+                        BankAPIJsonMapper.SINGLE_VALIDATOR_ENDPOINT -> {
+                            val content = BankAPIJsonMapper.mapValidatorToJson()
+                            sendResponse(content, errorContent, sendOnlyErrorResponses)
                         }
-                    }
-                    BankAPIJsonMapper.CONFIG_ENDPOINT -> {
-                        val content = BankAPIJsonMapper.mapBankDetailToJson()
-                        if (enableErrorResponse) {
-                            respond(errorContent, HttpStatusCode.InternalServerError, responseHeaders)
-                        } else {
-                            respond(content, HttpStatusCode.OK, responseHeaders)
+                        BankAPIJsonMapper.CONFIG_ENDPOINT -> {
+                            val content = BankAPIJsonMapper.mapBankDetailToJson()
+                            sendResponse(content, errorContent, sendOnlyErrorResponses)
+                        }
+                        else -> {
+                            error("Unhandled ${request.url.encodedPath}")
                         }
-                    }
-                    else -> {
-                        error("Unhandled ${request.url.encodedPath}")
                     }
                 }
             }
+
+            installJsonFeature()
         }
 
-        installJsonFeature()
+    private fun MockRequestHandleScope.sendResponse(
+        content: String,
+        errorContent: String,
+        isError: Boolean
+    ) = when {
+        isError -> respond(errorContent, InternalServerError, responseHeaders)
+        else -> respond(content, HttpStatusCode.OK, responseHeaders)
     }
 
-    private fun patchBankEngine(enableErrorResponse: Boolean = false, isInvalidResponse: Boolean = false) = HttpClient(MockEngine) {
+    private fun patchBankEngine(
+        enableErrorResponse: Boolean = false,
+        isInvalidResponse: Boolean = false
+    ) = HttpClient(MockEngine) {
         val errorContent = BankAPIJsonMapper.mapInternalServerErrorToJson()
 
         engine {
             addHandler { request ->
-                when (request.url.encodedPath) {
-                    BankAPIJsonMapper.BANKS_TRUST_ENDPOINT -> {
-                        val content = BankAPIJsonMapper.mapBankTrustResponseToJson()
+                when {
+                    request.url.encodedPath == BankAPIJsonMapper.BANKS_TRUST_ENDPOINT -> {
+                        val requestedTrust = readTrustFromRequest(request)
+                        val content = BankAPIJsonMapper.mapBankTrustResponseToJson(requestedTrust)
```

Comment:
@IanArb: I made this mock endpoint behave the same way now. Otherwise, the test case doesn't really test anything. You could have trust=42 in the request and then assert that trust=10 afterwards and it would have passed, because the 10 was hard-coded. That's what I tried to say on the other PR :)